src/comments/comment.service.ts

```diff
@@ -0,0 +1,360 @@
+      });
+      if (!question) {
+        throw new NotFoundException('Question not found');
+      }
+      const CommentQuestionShip = this.commentQuestionShipsRepository.create({
+        question: question,
+        comments: [comment],
+      });
+      await this.commentQuestionShipsRepository.save(CommentQuestionShip);
+    } else if (targetType === 'comment') {
+      const comment = await this.commentsRepository.findOneBy({ id: targetId });
+      if (!comment) {
+        throw new NotFoundException('Comment not found');
+      }
+      const subComment = [comment];
+      const CommentCommentShip = this.commentsRepository.create({
+        subComments: subComment,
+      });
+      await this.commentsRepository.save(CommentCommentShip);
+    }
+
+    const user = await this.usersService.getUserDtoById(userId);
+    if (!user) {
+      throw new NotFoundException(userId);
+    }
+    if (quoteUserId) {
+      const quoteUser = await this.usersService.getUserDtoById(quoteUserId);
+      if (!quoteUser) {
+        throw new NotFoundException(quoteUserId);
+      }
+      return {
+        id: comment.id,
+        commentableId: targetId,
+        commentableType: targetType,
+        content: content,
+        user: user,
+        created_at: comment.createdAt.getTime(),
+        agree_type: 0,
+        agree_count: 0,
+        disagree_count: 0,
+        quote: {
+          quote_id: quoteId,
+          quote_user: quoteUser,
+        },
+      };
+    } else {
+      return {
+        id: comment.id,
+        commentableId: targetId,
+        commentableType: targetType,
+        content: content,
+        user: user,
+        created_at: comment.createdAt.getTime(),
+        agree_type: 0,
+        agree_count: 0,
+        disagree_count: 0,
+        quote: {
+          quote_id: null,
+          quote_user: null,
+        },
+      };
+    }
+  }
+
+  async getCommentsById(commentId: number): Promise<CommentDto> {
+    const comment = await this.commentsRepository.findOneBy({ id: commentId });
+
+    if (!comment) {
+      throw new NotFoundException(`Comment with ID ${commentId} not found`);
+    }
+
+    const commentDto: CommentDto = {
+      id: comment.id,
+      commentableId: comment.commentableId,
+      commentableType: comment.commentableType,
+      content: comment.content,
+      user: comment.user,
+      created_at: comment.createdAt.getTime(),
+      agree_type: 0,
+      agree_count: 0,
+      disagree_count: 0,
+      quote: {
+        quote_id: null,
+        quote_user: null,
+      },
+    };
+
+    return commentDto;
+  }
+
+  async deleteComment(commentId: number): Promise<void> {
+    const comment = await this.commentsRepository.findOneBy({ id: commentId });
+
+    if (!comment) {
+      throw new NotFoundException(`Comment with id ${commentId} not found`);
+    }
+
+    await this.commentsRepository.remove(comment);
+  }
+
+  async commentAnswer(answerId: number, content: string): Promise<CommentDto> {
```

Comment:
commentAnswer is also user related... You need to save who create this comment. It's quite obvious